base/multidimensional.jl

```diff
@@ -1006,21 +1012,58 @@ julia> diff(vec(a))
   4
  -2
  12
+
+julia> diff(a, dims=2, prepend = 0)
+2×2 Matrix{Int64}:
+ 0   2
+ 0  10
```

Comment:
Thanks for the review and comments. I do actually use `np.diff` quite often, but it always start with zero. So, I've never thought if its prepend before or after diff-ing. Here, it would be nice to have similar behaviour to `np.diff` since so many people are going back and forth between Julia and Python. I'll update the PR accordingly.